examples/nlp/dialogue/generate_description_for_zero_shot_slot_filling.py

```diff
@@ -0,0 +1,139 @@
+                all_slots_keywords[k] = {}
+            else:
+                new_id = int((k - 1) / 2) + 1
+                merge_dict_slots[new_id] = full_name[v[0].split('-')[1]]
+                if dataset_name == 'conll_2003':
+                    merge_dict_slots[new_id] = [full_name[v[0].split('-')[1]]]
+                else:
+                    merge_dict_slots[new_id] = v
+                all_slots_keywords[new_id] = {}
+    else:
+        for k, v in dict_slots.items():
+            all_slots_keywords[k] = {}
+        merge_dict_slots = dict_slots
+    return merge_dict_slots, all_slots_keywords
+
+
+def get_key_words(train, train_slots, all_slots_keywords, dataset_name):
+    if dataset_name == 'assistant':
+        label_for_empty_entity = '54'
+    else:
+        label_for_empty_entity = '0'
+    # all_slots_keywords #key: slot_it, value: words
+    for idx, sentence in enumerate(train):
+        if idx > 0:
+            for word, label in zip(sentence.split()[:-1], train_slots[idx - 1].split()):
+                if label != label_for_empty_entity:
+                    if dataset_name == 'assistant':
+                        new_label = int(label)
+                    else:
+                        new_label = int((int(label) - 1) / 2) + 1
+                    new_word = word.lower()
+                    if new_word not in all_slots_keywords[new_label]:
+                        all_slots_keywords[new_label][new_word] = 1
+                    else:
+                        all_slots_keywords[new_label][new_word] += 1
+    return all_slots_keywords
+
+
+def get_class_description(all_slots_keywords):
+    description_list = []
+    punc = '''!()-[]{};:'"\,<>./?@#$%^&*_~'''
+
+    for k, slots_keywords in all_slots_keywords.items():
+        #     print(merge_dict_slots[k])
+        if k == 0:
+            description_list.append("other\tother")
+        else:
+            sorted_keywords = sorted(slots_keywords.items(), key=lambda item: item[1], reverse=True)
+            one_slot_description = ' '.join(
+                list(set([k for k, v in sorted_keywords if v > 3] + [k for k, v in sorted_keywords[:5]]))
+            )
+            entity_label = merge_dict_slots[k][0].split("-")[-1]
+            for c in entity_label:
+                if c in punc:
+                    entity_label = entity_label.replace(c, " ")
+
+            description_list.append(entity_label + '\t' + one_slot_description)
+    return description_list
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--preprocess_file_path", help="The folder contains all the dataset ")
+    parser.add_argument(
+        "--dataset",
+        choices=['drive_through', 'assistant', 'conll_2003'],
+        default='assistant',
+        help="the type of dataset",
+    )
+    args = parser.parse_args()
+
+    print(args)
+    if os.path.exists(args.preprocess_file_path):
+        print('folder exist')
```

Comment:
please remove all the print statements (after you're done debugging), they will be quite distracting for the user.